pkg/kv/bulk/bulkpb/ingestion_performance_stats.go

```diff
@@ -87,9 +87,76 @@ func (s *IngestionPerformanceStats) Combine(other bulk.TracingAggregatorEvent) {
 	}
 }
 
-// Tag implements the TracingAggregatorEvent interface.
+// ProtoName implements the TracingAggregatorEvent interface.
 func (s *IngestionPerformanceStats) ProtoName() string {
-	return reflect.TypeOf(s).Elem().String()
+	return proto.MessageName(s)
+}
+
+func (s *IngestionPerformanceStats) ToText() []byte {
+	return []byte(s.String())
+}
+
+// String implements the stringer interface.
+func (s *IngestionPerformanceStats) String() string {
+	const mb = 1 << 20
+	var b strings.Builder
+	if s.Batches > 0 {
+		b.WriteString(fmt.Sprintf("num_batches: %d\n", s.Batches))
+		b.WriteString(fmt.Sprintf("num_batches_due_to_size: %d\n", s.BatchesDueToSize))
+		b.WriteString(fmt.Sprintf("num_batches_due_to_range: %d\n", s.BatchesDueToRange))
+		b.WriteString(fmt.Sprintf("split_retries: %d\n", s.SplitRetries))
+	}
+
+	if s.BufferFlushes > 0 {
+		b.WriteString(fmt.Sprintf("num_flushes: %d\n", s.BufferFlushes))
+		b.WriteString(fmt.Sprintf("num_flushes_due_to_size: %d\n", s.FlushesDueToSize))
+	}
+
+	if s.LogicalDataSize > 0 {
+		logicalDataSizeMB := float64(s.LogicalDataSize) / mb
+		b.WriteString(fmt.Sprintf("logical_data_size: %.2f MB\n", logicalDataSizeMB))
+
+		if !s.CurrentFlushTime.IsEmpty() && !s.LastFlushTime.IsEmpty() {
+			duration := s.CurrentFlushTime.GoTime().Sub(s.LastFlushTime.GoTime())
+			throughput := logicalDataSizeMB / duration.Seconds()
+			b.WriteString(fmt.Sprintf("logical_throughput: %.2f MB/s\n", throughput))
+		}
+	}
+
+	if s.SSTDataSize > 0 {
+		sstDataSizeMB := float64(s.SSTDataSize) / mb
+		b.WriteString(fmt.Sprintf("sst_data_size: %.2f MB\n", sstDataSizeMB))
+
+		if !s.CurrentFlushTime.IsEmpty() && !s.LastFlushTime.IsEmpty() {
+			duration := s.CurrentFlushTime.GoTime().Sub(s.LastFlushTime.GoTime())
+			throughput := sstDataSizeMB / duration.Seconds()
+			b.WriteString(fmt.Sprintf("sst_throughput: %.2f MB/s\n", throughput))
+		}
+	}
+
+	timeString(&b, "fill_wait", s.FillWait)
+	timeString(&b, "sort_wait", s.SortWait)
+	timeString(&b, "flush_wait", s.FlushWait)
+	timeString(&b, "batch_wait", s.BatchWait)
+	timeString(&b, "send_wait", s.SendWait)
+	timeString(&b, "split_wait", s.SplitWait)
+	b.WriteString(fmt.Sprintf("splits: %d\n", s.Splits))
+	timeString(&b, "scatter_wait", s.ScatterWait)
```

Comment:
I could see a case for having all the waits next to each other followed by all the counts. But I think what you have here (all split metrics followed by all scatter metrics) also make sense.

Reply:
done